src/main/java/lotto/utils/LottoCompare.java

```diff
@@ -0,0 +1,46 @@
+package lotto.utils;
+
+import java.util.Arrays;
+
+public enum LottoCompare {
+    FIRST("6개 일치 (2,000,000,000원)", 6, false, 2000000000),
+    SECOND("5개 일치, 보너스 볼 일치 (30,000,000원)", 5, true, 30000000),
+    THIRD("5개 일치 (1,500,000원)", 5, false, 1500000),
+    FOURTH("4개 일치 (50,000원)", 4, false, 50000),
+    FIFTH("3개 일치 (5,000원)", 3, false, 5000),
+    NONE("0", 0, false, 0);
+
+    private final String label;
+    private final int match;
+    private final boolean needBonus;
+    private final int prize;
+
+    LottoCompare(String label, int match, boolean needBonus, int prize) {
+        this.label = label;
+        this.match = match;
+        this.needBonus = needBonus;
+        this.prize = prize;
+    }
+
+    public int getPrize() {
+        return prize;
+    }
+
+    public String getLabel() {
+        return label;
+    }
+
+    public static LottoCompare getCompareResult(int match, boolean needBonus) {
+        if (match == SECOND.match && needBonus != SECOND.needBonus) {
+            return THIRD;
+        }
+        if (match < FIFTH.match) {
+            return NONE;
+        }
+        return Arrays.stream(LottoCompare.values())
+                .filter(i -> i.match == match)
+                .findAny()
+                .orElseThrow(IllegalStateException::new);
```

Comment:
match의 수와 일치하는 `enum` 객체가 없는 경우 예외를 발생시키고 계신데요.
윗줄의 `NONE` 객체 리턴과 합친다면 조금 더 간결한 코드가 될 것 같아요.
```suggestion
        return Arrays.stream(LottoCompare.values())
                .filter(i -> i.match == match)
                .findAny()
                .orElse(NONE);
```